static/app/components/events/featureFlags/setupIntegrationModal.tsx

```diff
@@ -0,0 +1,218 @@
+  state,
+  orgSlug,
+}: {
+  orgSlug: string | undefined;
+  state: State;
+}) {
+  const api = useApi();
+  const date = new Date().toISOString();
+
+  const createToken = async () =>
+    await api.requestPromise(`/organizations/${orgSlug}/org-auth-tokens/`, {
+      method: 'POST',
+      data: {
+        name: `${state.provider} Token ${date}`,
+      },
+    });
+
+  return {createToken};
+}
+
+export function SetupIntegrationModal<T extends Data>({
+  Header,
+  Body,
+  Footer,
+  closeModal,
+}: ModalRenderProps) {
+  const [state, setState] = useState<State>({
+    provider: 'LaunchDarkly',
+    url: undefined,
+  });
+  const {organization} = useLegacyStore(OrganizationStore);
+  const {createToken} = useGenerateAuthToken({state, orgSlug: organization?.slug});
+
+  const handleSubmit = useCallback(() => {
+    addSuccessMessage(t('Integration set up successfully'));
+    closeModal();
+  }, [closeModal]);
+
+  const ModalHeader = useCallback(
+    ({children: headerChildren}: {children: React.ReactNode}) => {
+      return (
+        <Header closeButton>
+          <h3>{headerChildren}</h3>
+        </Header>
+      );
+    },
+    [Header]
+  );
+
+  const ModalFooter = useCallback(() => {
+    return (
+      <Footer>
+        <StyledButtonBar gap={1}>
+          <LinkButton
+            priority="default"
+            href="https://docs.sentry.io/product/issues/issue-details/#feature-flags"
+            external
+          >
+            {t('Read Docs')}
+          </LinkButton>
+          <Button
+            priority="primary"
+            title={!defined(state.provider) && t('Required fields must be filled out.')}
+            onClick={() => handleSubmit()}
+            disabled={!defined(state.provider)}
+          >
+            {t('Done')}
+          </Button>
+        </StyledButtonBar>
+      </Footer>
+    );
+  }, [Footer, handleSubmit, state]);
+
+  const ModalBody = useCallback(
+    ({children: bodyChildren}: Parameters<ChildrenProps<T>['Body']>[0]) => {
+      return <Body>{bodyChildren}</Body>;
+    },
+    [Body]
+  );
+
+  const integrations = ['LaunchDarkly'];
+
+  return (
+    <Fragment>
+      <ModalHeader>{t('Set Up Feature Flag Integration')}</ModalHeader>
+      <ModalBody>
+        <SelectContainer>
+          <SelectField
+            label={t('Feature Flag Services')}
+            name="provider"
+            inline={false}
+            options={integrations.map(integration => ({
+              value: integration,
+              label: integration,
+            }))}
+            placeholder={t('Select a feature flag service')}
+            value={state.provider}
+            onChange={value => setState({...state, provider: value})}
+            flexibleControlStateSize
+            stacked
+            required
+          />
+          <WebhookButton
+            priority="default"
+            title={!defined(state.provider) && t('You must select a provider first.')}
+            onClick={async () => {
```

Comment:
Could we abstract this onClick handler to its own useCallback above?

Reply:
yup, will do